modules/tinymce/src/plugins/link/main/ts/core/Actions.ts

```diff
@@ -6,11 +8,11 @@ import VK from 'tinymce/core/api/util/VK';
 import * as OpenUrl from './OpenUrl';
 import * as Utils from './Utils';
 
-const getLink = (editor: Editor, elm: Node): HTMLAnchorElement | null =>
-  editor.dom.getParent<HTMLAnchorElement>(elm, 'a[href]');
+const getLinks = (editor: Editor) => editor.selection.isCollapsed() ?
+  Utils.getLinks(editor.dom.getParents(editor.selection.getStart())) as [HTMLAnchorElement] :
+  Utils.getLinksInSelection(editor.selection.getRng());
 
-const getSelectedLink = (editor: Editor): HTMLAnchorElement | null =>
-  getLink(editor, editor.selection.getStart());
+const getSelectedLink = (editor: Editor): HTMLAnchorElement | null => Arr.head(getLinks(editor)).getOrNull();
```

Comment:
If you look at where this is used, the strictly `null` return type was just a factor from `editor.dom.getParent`. Not strictly necessary.
```suggestion
const getSelectedLink = (editor: Editor): HTMLAnchorElement | undefined => getLinks(editor)[0];
```